lib/logstash/filters/dictionary/csv_file.rb

```diff
@@ -6,19 +6,9 @@ class CsvFile < File
 
     protected
 
-    def initialize_for_file_type
-      @io = StringIO.new("")
-      @csv = ::CSV.new(@io)
-    end
-
     def read_file_into_dictionary
-      # low level CSV read that tries to create as
-      # few intermediate objects as possible
-      # this overwrites the value at key
-      IO.foreach(@dictionary_path, :mode => 'r:bom|utf-8') do |line|
-        @io.string = line
-        k,v = @csv.shift
-        @dictionary[k] = v
+      ::CSV.open(@dictionary_path, 'r:bom|utf-8') do |csv|
+        csv.each { |row| k,v = row; @dictionary[k] = v }
```

Comment:
Doesn't this work as well?

```suggestion
        csv.each { |k,v| @dictionary[k] = v }
```

Reply:
How is this `each` implemented? can you check that this way to iterate over the file won't use more memory in big files (i.e. no eager loading)?

Reply:
that should not be a concern but I went ahead and got a large (~800M) .csv file ([from here](https://www.stats.govt.nz/large-datasets/csv-files-for-download/)):
`837576 Data8277.csv` 

was able to parse using a 100M heap:
`JRUBY_OPTS="-J-Xmx100m -J-Xms100m -Xcompile.invokedynamic" jruby read_csv.rb Data8277.csv`
```
jruby 9.3.4.0 (2.6.8) 2022-03-23 eff48c1ebf OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +indy +jit [x86_64-linux]
read 34959673 rows in 230.392895
```
```
jruby 9.2.20.0 (2.5.8) 2021-11-02 1a3255440b OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +indy +jit [linux-x86_64]
read 34959673 rows in 229.984735
```

... even with 24M (~ 10M seems to have been too little and failed with OoME) : 
`JRUBY_OPTS="-J-Xmx24m -J-Xms24m -Xcompile.invokedynamic" jruby read_csv.rb Data8277.csv`
```
jruby 9.2.20.0 (2.5.8) 2021-11-02 1a3255440b OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +indy +jit [linux-x86_64]
read 34959673 rows in 396.616264
```

*read_csv.rb* script used: 
```ruby
require 'csv'

puts RUBY_DESCRIPTION

start = Time.now
rows = 0
::CSV.open(ARGV.first, 'r:bom|utf-8') do |csv|
  csv.each { |row| k,v = row; rows += 1; puts rows if rows % 100_000 == 0 }
end

puts RUBY_DESCRIPTION
puts "read #{rows} rows in #{Time.now - start}"
```

Reply:
Thanks for confirming !